tests/network/libs/connectivity.py

```
@@ -66,3 +75,38 @@ def poll_tcp_connectivity(
+
+    Raises:
+        IpNotFound: If no IPv4 address is found or console output cannot be parsed.
+    """
+    cmd: Final[str] = f"ip -j -4 addr show {interface_name}"
+    output = vm_console_run_commands(vm=vm, commands=[cmd], timeout=30)
+    LOGGER.info(f"Command {cmd} output: {output[cmd]}")
+
+    try:
+        iface_info = json.loads(output[cmd][1])
+    except json.JSONDecodeError as err:
+        raise IpNotFound(f"Failed to parse console JSON from VM {vm.name} for '{cmd}': {output[cmd]}") from err
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)